quarkus/addons/tracing-decision/integration-tests/src/test/java/org/kie/kogito/tracing/QuarkusTracingAddonDevServicesIT.java

```
@@ -0,0 +1,123 @@
+
+        await()
+                .atMost(30, TimeUnit.SECONDS)
+                .pollInterval(500, TimeUnit.MILLISECONDS)
+                .until(() -> {
+                    final Response response = given().when().get(trustyServiceEndpoint + "/executions");
+                    final String json = response.prettyPrint();
+                    return executionIds.stream().allMatch(executionId -> json.contains("\"executionId\": \"" + executionId + "\""));
+                });
+    }
+
+    private Response execute() {
+        return given()
+                .body("{" +
+                        "    \"Client\": {" +
+                        "        \"age\": 43," +
+                        "        \"salary\": 1950," +
+                        "        \"existing payments\": 100" +
+                        "    }," +
+                        "    \"Loan\": {" +
+                        "        \"duration\": 15," +
+                        "        \"installment\": 180" +
+                        "    }," +
+                        "    \"SupremeDirector\" : \"Yes\"," +
+                        "    \"Bribe\": 1000" +
+                        "}")
+                .contentType(ContentType.JSON)
+                .when()
+                .post("/LoanEligibility");
+    }
+
+    private String executeAndGetExecutionId() {
+        final Headers headers = execute().headers();
+
+        assertTrue(headers.hasHeaderWithName(KOGITO_EXECUTION_ID_HEADER));
+        return headers.getValue(KOGITO_EXECUTION_ID_HEADER);
+    }
+
+    private static String loadResource(final String name) {
+        try {
+            final URL url = QuarkusTracingAddonDevServicesIT.class.getResource(name);
+            if (Objects.nonNull(url)) {
+                return Files.readString(Paths.get(url.toURI()), StandardCharsets.UTF_8);
+            }
+        } catch (IOException | URISyntaxException e) {
+            // Swallow as we through an exception if the resource cannot be read
```

Shouldn't we rather fail fast?

---

I'm not sure. I would rather add "throw new IllegalStateException(String.format("Unable to load '%s'", name), e);" also in catch block as in general I don't like swallowed exceptions.